Content.Server/Administration/NetworkResourceManager.cs

```diff
@@ -0,0 +1,99 @@
+using Content.Server.Administration.Managers;
+using Content.Server.Database;
+using Content.Shared.Administration;
+using Content.Shared.CCVar;
+using Robust.Server.Player;
+using Robust.Shared.Configuration;
+using Robust.Shared.Network;
+
+namespace Content.Server.Administration;
+
+public sealed class NetworkResourceManager : SharedNetworkResourceManager
+{
+    [Dependency] private readonly IPlayerManager _playerManager = default!;
+    [Dependency] private readonly IAdminManager _adminManager = default!;
+    [Dependency] private readonly IServerNetManager _serverNetManager = default!;
+    [Dependency] private readonly IConfigurationManager _cfgManager = default!;
+    [Dependency] private readonly IServerDbManager _serverDb = default!;
+
+    [ViewVariables] public bool Enabled { get; private set; } = true;
+    [ViewVariables] public float SizeLimit { get; private set; } = 0f;
+    [ViewVariables] public bool StoreUploaded { get; set; } = true;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        _serverNetManager.Connected += ServerNetManagerOnConnected;
+        _cfgManager.OnValueChanged(CCVars.ResourceUploadingEnabled, value => Enabled = value, true);
+        _cfgManager.OnValueChanged(CCVars.ResourceUploadingLimitMb, value => SizeLimit = value, true);
+        _cfgManager.OnValueChanged(CCVars.ResourceUploadingStoreEnabled, value => StoreUploaded = value, true);
+
+        AutoDelete(_cfgManager.GetCVar(CCVars.ResourceUploadingStoreDeletionDays));
+    }
+
+    /// <summary>
+    ///     Callback for when a client attempts to upload a resource.
+    /// </summary>
+    /// <param name="msg"></param>
+    /// <exception cref="NotImplementedException"></exception>
+    protected override async void ResourceUploadMsg(NetworkResourceUploadMessage msg)
+    {
+        // Do not allow uploading any new resources if it has been disabled.
+        // Note: Any resources uploaded before being disabled will still be kept and sent.
+        if (!Enabled)
+            return;
+
+        if (!_playerManager.TryGetSessionByChannel(msg.MsgChannel, out var session))
+            return;
+
+        // +QUERY only for now.
+        if (!_adminManager.IsAdmin(session) || !_adminManager.HasAdminFlag(session, AdminFlags.Query))
```

Comment:
`HasAdminFlag` already checks whether they're admin.

```suggestion
        if (!_adminManager.HasAdminFlag(session, AdminFlags.Query))
```